### PR TITLE
fix(schematics): fixed schematics for ngx-bootstrap

### LINKED ISSUE #6078: Fix schematics for ng 11
FIX for nx 11 (build and test schematics)

Suggested fix - https://github.com/ngneat/lib#readme
or ng-bs or material

### LINKED ISSUE #6168: fix(schematics): add functionality for ng add for separate modules
add functionality for ng add for separate modules

### PR DESCRIPTION
**Pull request by SvetlanaMuravlova**
fix(schematics): fixed schematics for ngx-bootstrap
refs #6078 #6168 #6037

### CODE REVIEW

File: angular.json [hunk 1]
```diff
@@ -1286,15 +1287,18 @@
         "build": {
           "builder": "@nrwl/node:package",
           "outputs": [
-            "{options.outputPath}"
+            "dist/schematics"
```
Comment (valorkin) [inline angular.json]: I think you should be able to revert this line

File: angular.json [hunk 2]
```diff
@@ -1286,15 +1287,18 @@
...
             "outputPath": "dist/schematics",
             "tsConfig": "schematics/tsconfig.lib.json",
             "packageJson": "schematics/package.json",
-            "main": "schematics/src/index.ts",
+            "main": "schematics/src/ng-add/index.ts",
             "assets": [
-              "schematics/*.md"
+              "schematics/*.md",
+              "schematics/src/collection.json",
+              { "glob": "schema.json", "input": "schematics/src/ng-add", "output": "ng-add" }
```
Comment (valorkin) [inline angular.json:1300]:
```suggestion
              "schematics/**/schema.json"
```

```

File: schematics/src/ng-add/index.ts [hunk 3]
```diff
@@ -6,18 +6,22 @@
  * found in the LICENSE file at https://angular.io/license
  */
 
-import { ProjectDefinition, WorkspaceDefinition } from '@angular-devkit/core/src/workspace';
+import { WorkspaceDefinition } from '@angular-devkit/core/src/workspace';
```
Comment (valorkin) [inline schematics/src/ng-add/index.ts:9]: try
```suggestion
import { WorkspaceDefinition } from '@angular-devkit/core';
```

File: schematics/src/utils/index.ts [hunk 4]
```diff
@@ -41,25 +43,20 @@ export function addStyleToTarget(project: ProjectDefinition, targetName: string,
...
 }
 
-export function getProjectFromWorkspace(workspace: WorkspaceDefinition, projectName?: string): ProjectDefinition {
-
-  const _projectName = projectName || workspace.extensions.defaultProject?.toString();
-  const project: ProjectDefinition = workspace.projects.get(_projectName);
-
-  if (!project) {
-    throw new Error(`Could not find project in workspace: ${projectName}`);
+export function getProjectTargetOptions(project: WorkspaceProject, buildTarget: string) {
```
Comment (valorkin) [inline schematics/src/utils/index.ts]: why `ProjectDefinition` removed?

File: schematics/src/utils/index.ts [hunk 5]
```diff
@@ -41,25 +43,20 @@ export function addStyleToTarget(project: ProjectDefinition, targetName: string,
...
+  if (project.targets) {
+    const targetConfig = project.targets.get(buildTarget);
+    if (targetConfig && targetConfig.options) {
```
Comment (valorkin) [inline schematics/src/utils/index.ts]: same as in `project.architect'

File: scripts/schematics/copy-static-files.js [hunk 6]
```diff
@@ -0,0 +1,10 @@
+'use strict';
```
Comment (valorkin) [inline scripts/schematics/copy-static-files.js]: remove this file

**Review by valorkin — CHANGES_REQUESTED** [review]

File: angular.json (hunk 1 above)
Comment (SvetlanaMuravlova) [reply angular.json]: done

File: schematics/src/utils/index.ts (hunk 4 above)
Comment (SvetlanaMuravlova) [reply schematics/src/utils/index.ts]: done

File: schematics/src/utils/index.ts (hunk 5 above)
Comment (SvetlanaMuravlova) [reply schematics/src/utils/index.ts]: done

File: scripts/schematics/copy-static-files.js (hunk 6 above)
Comment (SvetlanaMuravlova) [reply scripts/schematics/copy-static-files.js]: done

File: schematics/src/ng-add/index.ts [hunk 7]
```diff
@@ -6,18 +6,22 @@
...
+  addStyleToTarget,
+  getProjectFromWorkSpace,
+  getWorkspace
+} from '../utils';
 import { hasNgModuleImport } from '../utils/ng-module-imports';
 import { getProjectMainFile } from '../utils/project-main-file';
 import { Schema } from './schema';
-
+import { WorkspaceProject } from '@schematics/angular/utility/workspace-models';
 const NGX_BOOTSTRAP_VERSION = '^6.2.0';
```
Comment (valorkin) [inline schematics/src/ng-add/index.ts]: should read from package.json maybe? create a separate issue, may require `version` script update

**Review by valorkin — DISMISSED** [review]

File: schematics/src/ng-add/index.ts (hunk 7 above)
Comment (SvetlanaMuravlova) [reply schematics/src/ng-add/index.ts]: done issue #6173

**Pull request merged**